agent/pool/pool.go

```diff
@@ -132,6 +175,9 @@ type ConnPool struct {
 	// TODO: consider refactoring to accept a full yamux.Config instead of a logger
 	Logger *log.Logger
 
+	// The default timeout for stream reads/writes
+	Timeout time.Duration
+
```

Comment:
Could you help me understand the difference between `ConnPool.Timeout` vs the timeout args that have been added to all the `rpc` methods?

I'm wondering if it's possible to achieve the same behaviour without adding an arg (which is defaulted to 0 in most cases) to an already long list of arguments to `rpc`.

Reply:
`ConnPool.Timeout` ('The default timeout for stream reads/writes') is set from `RPCHoldTimeout`, and represents the default timeout for any stream read/write when an explicit timeout for a specific read/write call is not provided. `RPCHoldTimeout` represents an existing duration the server is tuned to expect round-trips to be shorter than, so it seemed like a reasonable timeout to use for most reads/writes going through this connection pool.

The `ConnPool.rpc` method sets a specific `timeout` for the first stream read after the query is sent. This is because the expected wait time for a (potentially blocking) query response can be much longer than the default round-trip timeout configuration.

> I'm wondering if it's possible to achieve the same behaviour without adding an arg (which is defaulted to 0 in most cases) to an already long list of arguments to rpc.

Thanks for the suggestion. Yes, Dropping the `timeout` argument by moving the RPC-timeout calculation from `Client.RPC` to `ConnPool.rpc` requires `MaxQueryTime` and `DefaultQueryTime` to be passed to `ConnPool`, but I think it does end up a bit cleaner that way.